Magento_Sales integration tests: Fix addresses in order_list fixture

evktalo opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios,
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->

Addresses are broken in the order_list.php integration test fixture.

Instead of setting billing address twice, set both billing address and shipping address.
Reset $billingAddress and $shippingAddress variables with newly created Address objects. Otherwise trying to access the order's address data gives an error. (e.g. 'Call to a member function on null' when trying to use $order->getBillingAddress()->getFirstname())

Additionally, improve code style slightly.

I don't think this affects the present integration tests that use this fixture, but when you try to use this fixture in a test that does use the addresses, it will not work.

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->

None.

### Manual testing scenarios
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->

1. Create an integration test that uses order_list fixture
2. Create order in the test
3. Call $order->getShipppingAddress()->getFirstname()

Without the fix, you will get a 'Call to a member function on null' error.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

einokeskitalovaimo committed:
Fix setting billing address twice. Instead, set shipping address also. Additionally, improve code style slightly.

magento-engcom-team commented:
Hi @evktalo. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

einokeskitalovaimo committed:
Reset addresses for each order. Otherwise they cannot be accessed.

einokeskitalovaimo committed:
Improve var tags.

dmanners committed:
Update Magento_Sales integrations tests to fix error when running RMA module tests
 - Use the order repository for saving,
 - Use guest customer,
 - Attach email address to order,

dmanners committed:
Set the customer address and is gust flag before setting the addresses against the order

dmanners commented on dev/tests/integration/testsuite/Magento/Sales/_files/order_list.php:
```diff
@@ -48,16 +49,30 @@
         ->setData($orderData)
         ->addItem($orderItem)
+        ->setCustomerIsGuest(true)
+        ->setCustomerEmail('customer@null.com')
```
@evktalo so we had some issues with an API test because the order needed a valid email attached to it to save. I also refactored the code a bit to use the `$orderRepository` when saving. This makes it make the file `dev/tests/integration/testsuite/Magento/Sales/_files/order.php`

magento-engcom-team commented:
Hi @dmanners, thank you for the review.
[ENGCOM-2252](https://jira.corp.magento.com/browse/ENGCOM-2252) has been created to process this Pull Request

magento-engcom-team commented:
Hi @evktalo. Thank you for your contribution. 
We will aim to release these changes as part of 2.3.1. 
Please check the release notes for final confirmation.
